internal/pkg/cli/storage_init.go

```diff
@@ -692,65 +692,70 @@ func (o *initStorageOpts) readWorkloadType() error {
 }
 
 type addonBlob struct {
-	name        string
 	description string
 	blob        encoding.BinaryMarshaler
+	path        string
+	isEnvLevel  bool
 }
 
 func (o *initStorageOpts) addonBlobs() ([]addonBlob, error) {
-	templateBlob, err := o.newAddonTemplate()
+	type option struct {
+		lifecycle   string
+		storageType string
+	}
+	selection := option{o.lifecycle, o.storageType}
+	switch selection {
+	case option{lifecycleWorkloadLevel, s3StorageType}:
+		return o.wkldS3AddonBlobs()
+	case option{lifecycleWorkloadLevel, dynamoDBStorageType}:
+		return o.wkldDDBAddonBlobs()
+	case option{lifecycleWorkloadLevel, rdsStorageType}:
+		return o.wkldRDSAddonBlobs()
+	case option{lifecycleEnvironmentLevel, s3StorageType}:
+		return o.envS3AddonBlobs()
+	case option{lifecycleEnvironmentLevel, dynamoDBStorageType}:
+		return o.envDDBAddonBlobs()
+	case option{lifecycleEnvironmentLevel, rdsStorageType}:
+		return o.envRDSAddonBlobs()
+	}
+	return nil, fmt.Errorf("storage type %s is not supported yet", o.storageType)
+}
+
+func (o *initStorageOpts) wkldDDBAddonBlobs() ([]addonBlob, error) {
+	props, err := o.ddbProps()
 	if err != nil {
 		return nil, err
 	}
-	blobs := []addonBlob{
+	return []addonBlob{
 		{
-			name:        o.storageName,
+			path:        o.ws.WorkloadAddonFilePath(o.workloadName, fmt.Sprintf("%s.yml", o.storageName)),
 			description: "template",
-			blob:        templateBlob,
+			blob:        addon.WorkloadDDBTemplate(props),
 		},
-	}
-	paramsBlob, err := o.newAddonParams()
-	if err != nil {
-		if errors.Is(err, errUnavailableAddonParams) { // The addon does not need any parameters.
-			return blobs, nil
-		}
-		return nil, err
-	}
-	return append(blobs, addonBlob{
-		name:        "addons.parameters",
-		description: "parameters",
-		blob:        paramsBlob,
-	}), nil
+	}, nil
 }
 
-func (o *initStorageOpts) newAddonTemplate() (encoding.BinaryMarshaler, error) {
-	var templateBlob encoding.BinaryMarshaler
-	err := fmt.Errorf("storage type %s doesn't have a CF template", o.storageType)
-	switch o.storageType {
-	case dynamoDBStorageType:
-		templateBlob, err = o.newDDBTemplate()
-	case s3StorageType:
-		templateBlob, err = o.newS3Template()
-	case rdsStorageType:
-		templateBlob, err = o.newRDSTemplate()
-	}
+func (o *initStorageOpts) envDDBAddonBlobs() ([]addonBlob, error) {
+	props, err := o.ddbProps()
 	if err != nil {
 		return nil, err
 	}
-	return templateBlob, nil
-}
-
-func (o *initStorageOpts) newAddonParams() (encoding.BinaryMarshaler, error) {
-	if o.storageType != rdsStorageType {
-		return nil, errUnavailableAddonParams
-	}
-	if o.workloadType != manifest.RequestDrivenWebServiceType {
-		return nil, errUnavailableAddonParams
-	}
-	return addon.RDWSParamsForRDS(), nil
+	return []addonBlob{
+		{
+			path:        o.ws.EnvAddonFilePath(fmt.Sprintf("%s.yml", o.storageName)),
+			description: "template",
+			blob:        addon.EnvDDBTemplate(props),
+			isEnvLevel:  true,
+		},
+		{
```

Comment:
Since in the recommended action we will suggest for them to run `storage init` with a different flag in their workspace,

I wonder if we can simplify and remove the `isWritable` method if we just don't append the workload blob if `!workloadExists`

Reply:
done!